src/code_7bb60_len_41b0.c

```diff
@@ -10,7 +13,40 @@ void func_800E26B0(void) {
     playerStatus->jumpApexHeight = playerStatus->position.y;
 }
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E26C4);
+s32 func_800E26C4(void) {
+    PlayerStatus* playerStatus = gPlayerStatusPtr;
+    PlayerData* playerData = &gPlayerData;
+    s32 actionState = playerStatus->actionState;
+    s8* temp_8010EBB0 = &D_8010EBB0;
+
+    if (actionState == ActionState_IDLE ||
+        actionState == ActionState_WALK ||
+        actionState == ActionState_RUN ||
+        actionState == ActionState_USE_TWEESTER ||
+        actionState == ActionState_SPIN) {
+        return 1;
+    }
+
+    if (actionState == ActionState_RIDE) {
+        if (playerData->currentPartner == PartnerID_LAKILESTER || playerData->currentPartner == PartnerID_BOW) {
+            if (temp_8010EBB0[0] != 0) {
+                return 1;
+            } else {
+                playerStatus->animFlags |= 4;
+                return 0;
+            }
+        } else {
+            if ((u32)((u8)temp_8010EBB0[3] - 6) < 2) { // or of things like the partners above
```

Comment:
switch

Reply:
It's an or, not a switch. We learned that with the partnerID thing. Whenever it does a subtraction and then < some number and then an unsigned cast

Reply:
I made it into an or and it works...not sure why I didn't do that before